File: apps/pos/native-windows/DesktopKitchenPOS/Services/MenuService.cs
using DesktopKitchenPOS.Models;
using DesktopKitchenPOS.Networking.Endpoints;

namespace DesktopKitchenPOS.Services;

public class MenuService
{
    public async Task<List<MenuCategory>> GetCategoriesAsync()
    {
        return await DesktopKitchenPOS.App.Api.RequestAsync<List<MenuCategory>>(
            HttpMethod.Get, MenuEndpoints.Categories);
    }

    public async Task<List<MenuItem>> GetMenuItemsAsync(int? categoryId = null)
    {
        var path = categoryId.HasValue
            ? MenuEndpoints.ItemsByCategory(categoryId.Value)
            : MenuEndpoints.Items;
        return await DesktopKitchenPOS.App.Api.RequestAsync<List<MenuItem>>(HttpMethod.Get, path);
    }
}

File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/services/MenuService.kt
package com.desktopkitchen.pos.services

import com.desktopkitchen.pos.models.MenuCategory
import com.desktopkitchen.pos.models.MenuItem
import com.desktopkitchen.pos.networking.api.MenuApi
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class MenuService @Inject constructor(
    private val menuApi: MenuApi
) {
    suspend fun getCategories(): List<MenuCategory> = menuApi.getCategories()

    suspend fun getMenuItems(categoryId: Int? = null): List<MenuItem> = menuApi.getItems(categoryId)
}
